odiglet/pkg/instrumentation/fs/copy.go

```diff
@@ -85,28 +85,25 @@ func worker(fileChan <-chan string, sourceDir, destDir string, wg *sync.WaitGrou
 	}
 }
 
-func getFiles(dir string, CopyCFiles bool) ([]string, error) {
+func getFiles(dir string, CopyCFiles bool, filesToKeepMap map[string]struct{}) ([]string, error) {
 	var files []string
 	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
 		if !d.IsDir() {
-
 			if !CopyCFiles {
-				// filter out C files in ebpf directories
-				if strings.Contains(filepath.Dir(path), "ebpf") {
-					switch ext := filepath.Ext(path); ext {
-					case ".so", ".node", ".node.d", ".a":
-						return nil
-					}
+				if _, found := filesToKeepMap[strings.Replace(path, "/instrumentations/", "/var/odigos/", 1)]; found {
```

Comment:
Can we avoid calling `strings.Replace` on every file by saving everything in `filesToKeepMap` as `/instrumentations/.....` instead of `/var/odigos/....`

Reply:
Hey @edeNFed,
I'm using also in `removeFilesInDir` (which remove it from the hostDir which is /var/odigos)

Reply:
OK Got it